src/Algebra/Properties/Semigroup.agda

```diff
@@ -0,0 +1,122 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some theory for Semigroup
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra using (Semigroup)
+open import Algebra.FunctionProperties using (Commutative)
```

Comment:
Please follow the style guide and put all possible imports inside the module

Reply:
Done.

But consider the following.
1) Some import is often needed before. For example, Semigroup needs here to
   be imported before the head parameterized module.
2) The head module often has parameters.
   Setting all possible imports before this module stresses that these imports
   do not depend on these parameters. This brings more clearness.

Do you agree to withdraw the requirement?

Reply:
I agree with you that it's much clearer to have the module imports that don't depend on the module arguments above the header. However once again I'm afraid it's convention in the standard library to do it this way. I believe consistency is more useful then having different files follow different patterns. 

If you want to go through and change every module in the library at once (a lot of hard work!) then please do feel free (in a separate PR!) otherwise best to leave it.